src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs

```diff
@@ -46,6 +65,69 @@ public void Configure(ScriptJobHostOptions options)
             options.TestDataPath = webHostOptions.TestDataPath;
         }
 
+        private void ConfigureFunctionTimeout(IConfigurationSection jobHostSection, ScriptJobHostOptions options)
+        {
+            string value = jobHostSection.GetValue<string>("functionTimeout");
+            if (value != null)
+            {
+                TimeSpan requestedTimeout = TimeSpan.Parse(value, CultureInfo.InvariantCulture);
+
+                // Only apply limits if this is Dynamic.
+                if (_environment.IsDynamic() && (requestedTimeout < MinFunctionTimeout || requestedTimeout > MaxFunctionTimeout))
+                {
+                    string message = $"{nameof(options.FunctionTimeout)} must be between {MinFunctionTimeout} and {MaxFunctionTimeout}.";
+                    throw new ArgumentException(message);
+                }
+
+                options.FunctionTimeout = requestedTimeout;
+            }
+            else if (_environment.IsDynamic())
+            {
+                // Apply a default if this is running on Dynamic.
+                options.FunctionTimeout = DefaultFunctionTimeout;
+            }
+
+            // TODO: DI: JobHostOptions need to me updated.
+            //scriptConfig.HostOptions.FunctionTimeout = ScriptHost.CreateTimeoutConfiguration(scriptConfig);
```

Comment:
Opening this to track:
https://github.com/Azure/azure-functions-host/issues/3256